### PR TITLE
Updates chef-vault docs

### PR DESCRIPTION
**Pull request by kagarmoe**

### CODE REVIEW

**Comment by brewn** [comment]
This is awesome! 
tenor-47994578

File: chef_master/source/chef_vault.rst [hunk 1]
```diff
@@ -1,25 +1,558 @@
...
+   Chef vault requires the use of chef-client configured to use public/private key pairs. Chef vault is incompatible with the practice of using chef-client with a private key as ``client.pem`` and a certificate set as its public identity in the Chef server database. To update existing nodes to use chef-vault, first re-register your chef-client nodes with the Chef server, which will generate public/private key pairs, and then install Chef vault on each node. Chef vault will generate the following error if used with a chef-client with a private key as ``client.pem`` and a certificate set as its public identity in the Chef server database:
```
Comment (brewn) [inline chef_master/source/chef_vault.rst]: "Chef vault is incompatible with the practice of using chef-client with a private key as ``client.pem`` and a certificate set as its public identity in the Chef server database."

Not yours, but this sentence is unclear to me.

File: chef_master/source/chef_vault.rst [hunk 2]
```diff
@@ -1,25 +1,558 @@
 =====================================================
-chef-vault
+``chef-vault``
```
Comment (brewn) [inline chef_master/source/chef_vault.rst:2]: This is probably worth discussing, but I don't think we should use code preview syntax on titles. 

Also, we have three different versions of the chef-vault title in this guide: Chef vault, `chef-vault`, and chef-vault (with no code preview highlighting).

File: chef_master/source/chef_vault.rst [hunk 3]
```diff
@@ -1,25 +1,558 @@
...
+--------------------------------------
+
+.. code-block:: shell
+   
+   knife vault SUBCOMMAND VAULT ITEM VALUES
+
+where:
+
+* ``vault`` names the location for storing the encrypted item.
+* ``item`` names the item stored the vault.
```
Comment (brewn) [inline chef_master/source/chef_vault.rst]: stored *in* the vault.

File: chef_master/source/chef_vault.rst [hunk 4]
```diff
@@ -1,25 +1,558 @@
...
+      API Client Username
+
+``--print-after``
+      Show the data after a destructive operation
+
+``-S``, ``--search SEARCH``
+      Chef SOLR search for clients
+
+``-M``, ``--mode MODE``
+      Chef mode to run in default - solo
```
Comment (brewn) [inline chef_master/source/chef_vault.rst]: if this is being supplied via the command line, `solo` should probably have the code preview syntax. It also might be better to use **Default Value:** as that's the standard we've gone with in resource documentation.

File: chef_master/source/chef_vault.rst [hunk 5]
```diff
@@ -1,25 +1,558 @@
...
+ ``-h``, ``--help``
+      Show this message
+
+Example Commands
+--------------------------------------------------
+
+``create``
+--------------------------------------------------
+      
+Create a vault called passwords and put an item called root in it with the given values for username and password encrypted for clients role:webserver, client1 & client2 and admins admin1 & admin2
```
Comment (brewn) [inline chef_master/source/chef_vault.rst:183]: I think these introductions should end with a colon.

File: chef_master/source/chef_vault.rst [hunk 6]
```diff
@@ -1,25 +1,558 @@
...
+
+      .. code-block:: shell
+
+         knife vault create passwords root -A "admin1,admin2"
+
+.. note:: A JSON file can be used in place of specifying the values on the command line, see global options below for details
+
+``update``
+--------------------------------------------------
+Update the values in username and password in the vault passwords and item root.  Will overwrite existing values if values already exist!
```
Comment (brewn) [inline chef_master/source/chef_vault.rst:233]: I think this first sentence could be a little more clear, not sure what it's doing.

File: chef_master/source/chef_vault.rst [hunk 7]
```diff
@@ -1,25 +1,558 @@
...
+
+``refresh``
+--------------------------------------------------
+This command reads the search_query in the vault item, performs the search, and reapplies the results.
+
+     .. code-block:: shell
+     
+        knife vault refresh VAULT ITEM
+
+To remove clients which have been deleted from Chef but not from the vault, add the --clean-unknown-clients switch:
```
Comment (brewn) [inline chef_master/source/chef_vault.rst]: `--clean-unknown-clients` and any command flags should probably use code preview syntax.,

File: chef_master/source/chef_vault.rst [hunk 8]
```diff
@@ -48,8 +581,132 @@ Use the following options with a validatorless bootstrap to specify items that a
...
+Use ``chef-vault --help`` to see all all available options
+
+Example usage (password)
+--------------------------------------------------
+    chef-vault -v passwords -i root -a password -k /etc/chef/knife.rb
+
+Testing
+----------------------------------------------------------------------------
+To stub vault items in ChefSpec, use the
+[chef-vault-testfixtures](https://rubygems.org/gems/chef-vault-testfixtures)
```
Comment (brewn) [inline chef_master/source/chef_vault.rst]: This is a markdown link.

File: chef_master/source/chef_vault.rst [hunk 9]
```diff
@@ -48,8 +581,132 @@ Use the following options with a validatorless bootstrap to specify items that a
...
+
+    case ChefVault::Item.data_bag_item_type('passwords', 'root')
+    when :normal
+      ...
+    when :encrypted
+      ...
+    when :vault
+    end
+
+This functionality is also available from the command line as `knife vault itemtype VAULT ITEM`.
```
Comment (brewn) [inline chef_master/source/chef_vault.rst]: substitute ``

File: chef_master/source/chef_vault.rst (hunk 1 above)
Comment (kagarmoe) [reply chef_master/source/chef_vault.rst]: I am hoping that it makes sense for the people who need it.

File: chef_master/source/chef_vault.rst (hunk 3 above)
Comment (kagarmoe) [reply chef_master/source/chef_vault.rst]: +

Comment (kagarmoe) [reply chef_master/source/chef_vault.rst]: ok

File: chef_master/source/chef_vault.rst (hunk 9 above)
Comment (kagarmoe) [reply chef_master/source/chef_vault.rst]: Ok

File: chef_master/source/chef_vault.rst (hunk 7 above)
Comment (kagarmoe) [reply chef_master/source/chef_vault.rst]: ok

**Pull request merged**